compiler/front/main.nim

```diff
@@ -428,6 +426,37 @@ proc `$`(params: UsedBuildParams): string =
       #[9]# suffix
     ]
 
+proc customizeForBackend*(graph: ModuleGraph, conf: ConfigRef,
+                          backend: TBackend) =
+  ## Sets backend specific options. This must be called by all commands.
+  ## `backend` is the backend to choose when no backend has been selected
+  ## so far.
+  if conf.backend == backendInvalid:
+    # only set if wasn't already set, to allow override via `nim c -b:js`
+    conf.backend = backend
+
+  defineSymbol(graph.config, $conf.backend)
+  case conf.backend
+  of backendC:
+    case conf.exc
+    of excNone, excNative: conf.exc = excGoto
+    of excGoto:            discard
+
+    if conf.selectedGC == gcUnselected:
+      # the default gc for the C backend is ORC. We can't just set it to
+      # ``gcOrc`` directly, however, as additional defines, etc. are required
+      # XXX: the dependency on ``optionsprocessor`` hints that a different
```

Comment:
I've held off on separating flag parsing vs flag "executing", the latter meaning actually acting on some strictly defined data describing the parsed out flag.

If those were separated then there could be a direct call to the executing part, skipping past this silly string business.